app/services/sign_in/session_refresher.rb

```diff
@@ -20,6 +21,12 @@ def perform
 
     private
 
+    def validate_credential_lock
+      if session.user_verification&.locked
```

Comment:
you don't need the safety operator `&.` here, you can assume the `user_verification` exists here